cmd/spire-server/cli/run/run.go

```diff
@@ -366,6 +366,12 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 
 	sc.DataDir = c.Server.DataDir
 
+	// Warn on a non-conforming trust domain to avoid breaking backwards compatibility
+	if len(c.Server.TrustDomain) > common_cli.MaxTrustDomainLength {
+		logger.Warnf("Configured trust domain should be less than %d characters to be SPIFFE compliant",
```

Comment:
is the idea to keep this backward compatibility for more than one release? may we suggest to update it because we'll avoid this in a future?

Reply:
Based on the comments from @azdagron it sounds like this may stick around for more than one release due to there being no easy way to migrate to a new trust domain.